src/account_manager.rs

```diff
@@ -41,67 +52,89 @@ pub struct AccountManager {
     /// the polling interval.
     #[getset(get = "pub", set = "pub")]
     polling_interval: Duration,
-    started_monitoring: bool,
+    accounts: AccountStore,
+    stop_polling_sender: Option<BroadcastSender<()>>,
 }
 
-/// Internal transfer response metadata.
-pub struct InternalTransferMetadata {
-    /// Transfer message.
-    pub message: Message,
-    /// Source account with new message and addresses attached.
-    pub from_account: Account,
-    /// Destination account with new message attached.
-    pub to_account: Account,
+impl Drop for AccountManager {
+    fn drop(&mut self) {
+        let accounts = self.accounts.clone();
+        let stop_polling_sender = self.stop_polling_sender.clone();
+        thread::spawn(move || {
+            let _ = crate::block_on(async {
+                for account in accounts.read().await.values() {
+                    let _ = crate::monitor::unsubscribe(account.clone());
+                }
+
+                if let Some(sender) = stop_polling_sender {
+                    sender.send(()).expect("failed to stop polling process");
+                }
+            });
+        });
```

Comment:
Just so you're aware, spawning a thread and not waiting for it to be joined opens up the possibility of the thread being aborted on program exit. I don't understand the ramifications of this in the context of your code, so this might not be an issue at all.